actors/runtime/src/main/java/cloud/orbit/actors/runtime/Hosting.java

```diff
@@ -254,10 +260,11 @@ private void onClusterViewChanged(final Collection<NodeAddress> nodes)
         consistentHashNodeTree = newHashes;
         updateServerNodes();
 
+        Collection<NodeAddress> values = localAddressCache.asMap().values();
         for (NodeInfo info : oldNodes.values())
         {
             // clear local cache
-            localAddressCache.values().remove(info.address);
+            values.remove(info.address);
```

Comment:
we would need to know how many are in the cache... we could simply change this line to...
while (values.remove(info.address)) {}